plugins/eddn.py

```diff
@@ -35,121 +46,147 @@
+#
+# 2, a good few of these methods are static or could be classmethods. they should be created as such.
 
-    ### SERVER = 'http://localhost:8081'	# testing
+class EDDN:
+    # SERVER = 'http://localhost:8081'	# testing
     SERVER = 'https://eddn.edcd.io:4430'
-    UPLOAD = '%s/upload/' % SERVER
-    REPLAYPERIOD = 400	# Roughly two messages per second, accounting for send delays [ms]
-    REPLAYFLUSH = 20	# Update log on disk roughly every 10 seconds
-    TIMEOUT= 10	# requests timeout
-    MODULE_RE = re.compile('^Hpt_|^Int_|Armour_', re.IGNORECASE)
+    UPLOAD = f'{SERVER}/upload/'
+    REPLAYPERIOD = 400  # Roughly two messages per second, accounting for send delays [ms]
+    REPLAYFLUSH = 20  # Update log on disk roughly every 10 seconds
+    TIMEOUT = 10  # requests timeout
+    MODULE_RE = re.compile(r'^Hpt_|^Int_|Armour_', re.IGNORECASE)
     CANONICALISE_RE = re.compile(r'\$(.+)_name;')
 
-    def __init__(self, parent):
-        self.parent = parent
+    def __init__(self, parent: tk.Tk):
+        self.parent: tk.Tk = parent
         self.session = requests.Session()
-        self.replayfile = None	# For delayed messages
-        self.replaylog = []
+        self.replayfile: Optional[TextIO] = None  # For delayed messages
+        self.replaylog: List[str] = []
 
-    def load(self):
+    def load(self) -> bool:
         # Try to obtain exclusive access to the journal cache
         filename = join(config.app_dir, 'replay.jsonl')
         try:
             try:
                 # Try to open existing file
                 self.replayfile = open(filename, 'r+', buffering=1)
+
             except Exception:
                 if exists(filename):
                     raise  # Couldn't open existing file
+
                 else:
                     self.replayfile = open(filename, 'w+', buffering=1)  # Create file
+
             if sys.platform != 'win32':  # open for writing is automatically exclusive on Windows
                 lockf(self.replayfile, LOCK_EX | LOCK_NB)
+
         except Exception as e:
             logger.debug('Failed opening "replay.jsonl"', exc_info=e)
             if self.replayfile:
                 self.replayfile.close()
+
             self.replayfile = None
             return False
+
         self.replaylog = [line.strip() for line in self.replayfile]
         return True
 
     def flush(self):
         self.replayfile.seek(0, SEEK_SET)
         self.replayfile.truncate()
         for line in self.replaylog:
-            self.replayfile.write('%s\n' % line)
+            self.replayfile.write(f'{line}\n')
+
         self.replayfile.flush()
 
     def close(self):
         if self.replayfile:
             self.replayfile.close()
+
         self.replayfile = None
 
-    def send(self, cmdr, msg):
-        uploaderID = cmdr
+    def send(self, cmdr: str, msg: Mapping[str, Any]):
+        uploader_id = cmdr
 
-        msg = OrderedDict([
+        to_send: OrderedDictT[str, str] = OrderedDict([
```

Comment:
Is this a case where OrderedDict() is still necessary ?

Reply:
In most of the code almost none of it is necessary. For the most part in cleanups Ive been leaving them just in case we end up somewhere thats doing an isinstance. In #660 I pretty much just replaced a bunch as I was controlling the code path there